### ISSUE TITLE
PipeMode all files under prefix

### ISSUE THREAD

**Issue by gdj0nes** [aws/sagemaker-tensorflow-extensions#24]
Is there a way to access all files under a prefix using PipeModel using a channel? If so it would be nice to add an example to the documentation since this is likely a common use case.

**Comment by owen-t** [comment]
If you're using the [SageMaker Python SDK](https://github.com/aws/sagemaker-python-sdk), then simply pass a S3 prefix URIs to your estimator's fit method.

**Comment by gdj0nes** [comment]
Thanks!

**Comment by owen-t** [comment]
I'm going to leave this issue open for the time being. The point about incorporating into the docs is a good one.

**Comment by gdj0nes** [comment]
Could you provide a short example of using PipeModeDataset with multiple files with multiple channels i.e. for training and evaluation

**Comment by kalpitsmehta** [comment]
Can you please provide an example of how to parse the Augmented Manifest file in the entry_point script so as to get image data (from the URL) and label?

**Comment by athewsey** [comment]
It might be a bit off-topic from the issue, but since @kalpitsmehta asked and somebody +1'd:

If your channel is in [Augmented Manifest](https://docs.aws.amazon.com/sagemaker/latest/dg/augmented-manifest.html) mode, I think per the docs you should receive the binary file contents (not the URL) for any manifest attributes ending in `-ref`.

For example if my manifest has `source-ref` (image URI) and `labels` (SM Ground Truth label data), I'll get alternating records of JPEG/PNG/whatever data, and label data.

So the first step is to batch your dataset by the number of manifest file attributes, and then you want to create a `map()` to process the raw (binary) string tensors into the data type you actually want.

TensorFlow has [decode_image()](https://www.tensorflow.org/versions/r1.15/api_docs/python/tf/io/decode_image) but sadly I haven't yet found any good functions for loading JSON data... 

As a workaround (with a performance hit due to the Python GIL), we can use a [tf.py_func()](https://www.tensorflow.org/versions/r1.15/api_docs/python/tf/py_func) inside the mapper: Inside the py_func you should be able to call json.loads() no problem, and just return whatever fragment of the data you want as a numpy array.

I think something along these lines should do it?:

```python
def py_parse_json(label):
    # Extracting a numeric array from some subfield:
    return np.array(json.loads(label)["annotations"]["etc"])

def tf_parse_sample(fields):
    img = tf.io.decode_image(fields[0])
    boxes = tf.py_func(py_parse_json, [fields[1]], [tf.float64])
    boxes.set_shape([None, 5])  # TF doesn't know shape of py_func outputs
    return (img, boxes)

ds_train = PipeModeDataset(channel="train") \
        .repeat(args.epochs) \
        .batch(2) \
        .map(tf_parse_sample)
```
